### DISCUSSION TITLE
Why do we need `raw.execute`?

### DISCUSSION THREAD

**Discussion by jordandsullivan** [unitaryfund/mitiq#2327] [Q&A] [answered]
If you just want to get an unmitigated expectation value for some relatively simple execute function, there appear to be two different equivalent ways of doing so: 

`raw.execute(circuit, executor, observable)`
vs. 
`observable.expectation(circuit, executor)`?

In the docstring for raw.execute, it states
https://github.com/unitaryfund/mitiq/blob/42779eda439778e821da513dcd1e21af94b86739/mitiq/raw/raw.py#L21-L24

But it appears other methods do not have a pure `execute` function. They have `execute_with_[qem_method]` and `mitigate_executor` functions, but I don't see how having a `raw.execute` function is useful in maintaining parity across QEM methods. 

It seems to me that it would be more logical to use `observable.expectation(circuit, executor)`, since you can presumably define any executor you wish, with or without error mitigation. To me, this would maintain a more consistent interface.

**Comment by jordandsullivan** [comment]
I also think it's not immediately obvious from the name that `raw.execute` will give you an expectation value rather than, say, measurement counts.

**Comment by cosenal** [comment, marked as answer]
All the context seems to be in https://github.com/unitaryfund/mitiq/issues/1006

I sympathize with your comments though, there is basically three ways to do the same thing, and only for documentation purposes.

**Comment by jordandsullivan** [comment]
Yeah personally I'm not a fan of this much redundancy,  since it makes the user waste time trying to figure out which API calls actually do something different vs. which ones are functionally an alias for another command.
